tests/slow/src/test/scala/tests/MillSlowSuite.scala

```diff
@@ -0,0 +1,239 @@
+          |package reload
+          |object Main extends App {
+          |  println("sourcecode.Line(42)")
+          |}
+          |""".stripMargin
+      )
+      _ <- server.didOpen("foo/src/reload/Main.scala")
+      _ = assertNoDiff(client.workspaceDiagnostics, "")
+      _ <- server.didSave("build.sc") { text =>
+        text.replace(
+          "/*DEPS*/",
+          "def ivyDeps = Agg(ivy\"com.lihaoyi::sourcecode::0.1.4\")"
+        )
+      }
+      _ <- server
+        .didSave("foo/src/reload/Main.scala") { text =>
+          text.replaceAll("\"", "")
+        }
+        .recover { case e => scribe.error("compile", e) }
+      _ = assertNoDiff(client.workspaceDiagnostics, "")
+    } yield ()
+  }
+
+  testAsync("cancel") {
+    client.slowTaskHandler = params => {
+      if (params == bloopInstallProgress("mill")) {
+        Thread.sleep(TimeUnit.SECONDS.toMillis(2))
+        Some(MetalsSlowTaskResult(cancel = true))
+      } else {
+        None
+      }
+    }
+    cleanWorkspace()
+    for {
+      _ <- server.initialize(
+        """
+          |/build.sc
+          |import mill._, scalalib._
+          |object foo extends ScalaModule {
+          |  def scalaVersion = "2.12.8"
+          |  /*DIFF*/
+          |}
+        """.stripMargin,
+        expectError = true
+      )
+      _ = assertStatus(!_.isInstalled)
+      _ = client.slowTaskHandler = _ => None
+      _ <- server.didSave("build.sc")(_ + "\n// comment \n")
+      _ = assertNoDiff(client.workspaceShowMessages, "")
+      _ = assertStatus(!_.isInstalled)
+      _ <- server.didSave("build.sc") { text =>
+        val a = text.replace("/*DIFF*/", "def test = 123"); println(a); a
+      }
+      _ = assertNoDiff(client.workspaceShowMessages, "")
+      _ = assertStatus(_.isInstalled)
+    } yield ()
+  }
+
+  testAsync("error") {
+    cleanWorkspace()
+    for {
+      _ <- server.initialize(
+        """|/build.sc
+           |, syntax error
+           |""".stripMargin,
+        expectError = true
+      )
+      _ = assertNoDiff(
+        client.workspaceMessageRequests,
+        List(
+          ImportBuild.params("mill").getMessage,
+          bloopInstallProgress("mill").message
+        ).mkString("\n")
+      )
+      _ = assertNoDiff(
+        client.workspaceShowMessages,
+        ImportProjectFailed.getMessage
+      )
+      _ = assertStatus(!_.isInstalled)
+      _ = client.messageRequests.clear()
+      _ <- server.didSave("build.sc") { _ =>
+        """
+          |import mill._, scalalib._
+          |object foo extends ScalaModule {
+          |  def scalaVersion = "2.12.8"
+          |}
+        """.stripMargin,
+      }
+      _ = assertNoDiff(
+        client.workspaceMessageRequests,
+        List(
+          ImportBuild.params("mill").getMessage,
+          bloopInstallProgress("mill").message
+        ).mkString("\n")
+      )
+      _ = assertStatus(_.isInstalled)
+    } yield ()
+  }
+
+  testAsync("fatal-warnings") {
```

Comment:
This test can be removed since we don't modify the build settings, it's failing right now.